### PR TITLE
Add new module for PagerDuty Events API v2

### PR DESCRIPTION
**Pull request by TheMonolithX64**
##### SUMMARY
Module to trigger, acknowledge and resolve PagerDuty incidents via Events API v2

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
pagerduty_event

##### ANSIBLE VERSION
```
ansible 2.7.0.dev0 (devel 3a9a77ee23) last updated 2018/06/10 20:11:03 (GMT +100)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/steve/sbdev/ansible/lib/ansible']
  ansible python module location = /home/steve/sbdev/ansible/lib/ansible
  executable location = /home/steve/sbdev/ansible/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
```

##### ADDITIONAL INFORMATION
There is an existing module [pagerduty_alert](https://docs.ansible.com/ansible/latest/modules/pagerduty_alert_module.html) which is using PagerDuty Events API v1. [PagerDuty v1 REST API is being decommisioned on Oct 19th](https://v2.developer.pagerduty.com/v2/docs/v1-rest-api-decommissioning-faq), and requires v1 API keys, which can no longer be generated. As such, this module will be an option for Ansible users.

Happy to discuss any consolidation and/or deprecation if deemed necessary.

### CODE REVIEW

**Comment by TheMonolithX64** [comment]
On thinking further about this PR, I decided to refactor to support both PagerDuty Events API v1 and v2. From the PagerDuty docs, nothing changes for the Events API on the REST v1 decommissioning.

As such, I would be proposing to deprecate pagerduty_alert.py module, which only support Events API v1.

**Comment by thaumos** [comment]
I am fine with the deprecation of the alert module.  I am no longer actively maintaining it.

File: lib/ansible/modules/monitoring/pagerduty_event.py [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import fetch_url
+
+
+def run_module():
+    # define the available arguments/parameters that a user can pass to
+    # the module
+    module_args = dict(
+        routing_key=dict(type='str', required=True, aliases=['service_key']),
+        state=dict(type='str', required=True,
```
Comment (thaumos) [inline lib/ansible/modules/monitoring/pagerduty_event.py:209]: these states are a little confusing...  granted I am not using pagerduty day to day anymore.

Is there a way to use our commonly used states (present, absent, etc)?

Comment (TheMonolithX64) [reply lib/ansible/modules/monitoring/pagerduty_event.py:209]: Thanks for the comments @thaumos! The PagerDuty Events API (v1 and v2) accepts the actions: `trigger`, `acknowledge` and `resolve` for event state. Hence, they do not correlate to the common Ansible module states.

I can change the current mapping of `triggered`, `acknowledged` and `resolved` to the slightly more shorthand version if required.

**Comment by TheMonolithX64** [comment]
@thaumos Thanks for the OK on deprecating pagerduty_alert.py. I am following the instructions [here](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_checklist.html#deprecating-and-making-module-aliases) regarding updating documentation. However, I cannot seem to find `CHANGELOG.md` in order to add to Deprecations section. 

Also, is there a specific version at which the deprecation will take place, for example 2.10? Any docs on that side of things?

File: lib/ansible/modules/monitoring/pagerduty_event.py (hunk 1 above)
Comment (TheMonolithX64) [reply lib/ansible/modules/monitoring/pagerduty_event.py:209]: Updated to use the shorthand states of `trigger`, `acknowledge` and `resolve` @thaumos

**Comment by samccann** [comment]
Hi @TheMonolithX64 -  Sorry for the delay. You can find changelog fragment details for documentation at:
https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs-how-to

You'll also need to rebase and update the release versions listed (porting guide is 2.9 now, for example).  Rebase details are at:
https://docs.ansible.com/ansible/latest/dev_guide/developing_rebasing.html

Thanks

**Pull request closed without merging**